packages/tradershub/src/hooks/useCurrencies.ts

```diff
@@ -80,11 +97,55 @@ const useCurrencies = () => {
         [currencyConfig?.CRYPTO]
     );
 
+    // Get the current account currency with its config and isAdded status
+    const currentAccountCurrencyConfig = useMemo(() => {
+        if (!authorizeData?.currency || !websiteStatusData?.website_status?.currencies_config) return;
+
+        return {
+            ...websiteStatusData?.website_status?.currencies_config[authorizeData?.currency],
+            id: authorizeData?.currency,
+        };
+    }, [authorizeData?.currency, websiteStatusData?.website_status?.currencies_config]);
+
+    // Get the added fiat currency with its config
+    const addedFiatCurrency = useMemo(() => {
+        return currencyConfig?.FIAT.find(currency => currency.isAdded);
+    }, [currencyConfig?.FIAT]);
+
+    // Disable fiat currencies if the current account currency is not fiat or if the account is deposit attempt
+    const disableFiatCurrencies = useMemo(
+        () =>
+            // if the current account currency is not fiat
+            (!!addedFiatCurrency && currentAccountCurrencyConfig?.type !== 'fiat') ||
+            // if there is balance in the account and the current account currency is fiat
+            (!!activeDerivTradingAccount?.balance && currentAccountCurrencyConfig?.type === 'fiat') ||
+            // if the account is deposit attempt and the current account currency is fiat
+            (currentAccountCurrencyConfig?.type === 'fiat' &&
+                accountStatus?.is_deposit_attempt &&
+                !!cfdAccountsList?.dxtrade && // if there is no dxtrade account
+                !!cfdAccountsList.mt5 && // if there is no mt5 account
+                statements?.statement?.count === 0 && // if there is no statement && transactions
```

Comment:
I believe you should negate this:
```suggestion
                statements?.statement?.count !== 0 && // if there is no statement && transactions
```

Reply:
I guess the comment needs to be updated
The condition is based on production implementation which I just noticed is a bit weird tho: 
`const has_no_transaction = this.statement.count === 0 && this.statement.transactions.length === 0;`

https://github.com/binary-com/deriv-app/blob/d82808bc492c6e26b1f903157f96503ff177d298/packages/core/src/Stores/client-store.js#L476